File: app/src/main/java/com/tobikster/medicreminder/data/ReminderDao.java
package com.tobikster.medicreminder.data;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface ReminderDao {
	@Query("SELECT * FROM reminder")
	Flowable<List<Reminder>> getAll();

	@Insert(onConflict = OnConflictStrategy.REPLACE)
	void insertReminder(final Reminder reminder);

	@Query("DELETE FROM reminder")
	void removeAllReminders();

}

File: app/src/main/java/com/tobikster/medicreminder/data/reminders/ReminderDao.kt
package com.tobikster.medicreminder.data.reminders

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query

@Dao
interface ReminderDao {
	@Query("SELECT * FROM Reminder")
	fun getAllReminders(): LiveData<List<Reminder>>

	@Query("SELECT * FROM Reminder WHERE id=:id")
	fun getReminder(id: Long): LiveData<Reminder>

	@Insert
	fun addReminder(reminder: Reminder): Long
}
